### ISSUE TITLE
About the GraphGPS model

### ISSUE THREAD

**Issue by kou18n** [pyg-team/pytorch_geometric#7438]
### 🐛 Describe the bug

Hello, this is my training data information.
`Data(edge_index=[2, 5187], y=[1], num_nodes=1611, x=[1611, 5], pe=[1611, 5], edge_attr=[5187])`
I plan to implement GPS model on other dataset.
For the ZINC dataset, the node feature x's dimension is one, everything is fine.
If the node feature dimension is 5, how should I define the model and forward function.

The example code (https://github.com/pyg-team/pytorch_geometric/blob/master/examples/graph_gps.py)

```python
class GPS(torch.nn.Module):
    def __init__(self, channels: int, pe_dim: int, num_layers: int):
        super().__init__()
       
        # change here?
        self.node_emb = Embedding(28, channels - pe_dim) 
        self.pe_lin = Linear(20, pe_dim)
        self.pe_norm = BatchNorm1d(20)
        self.edge_emb = Embedding(4, channels)

        self.convs = ModuleList()
        for _ in range(num_layers):
            nn = Sequential(
                Linear(channels, channels),
                ReLU(),
                Linear(channels, channels),
            )
            conv = GPSConv(channels, GINEConv(nn), heads=4, attn_dropout=0.5)
            self.convs.append(conv)

        self.mlp = Sequential(
            Linear(channels, channels // 2),
            ReLU(),
            Linear(channels // 2, channels // 4),
            ReLU(),
            Linear(channels // 4, 1),
        )

    def forward(self, x, pe, edge_index, edge_attr, batch):
        x_pe = self.pe_norm(pe)
        # change here?
        x = torch.cat((self.node_emb(x.squeeze(-1)), self.pe_lin(x_pe)), 1) 
        edge_attr = self.edge_emb(edge_attr)

        for conv in self.convs:
            x = conv(x, edge_index, batch, edge_attr=edge_attr)
        x = global_add_pool(x, batch)
        return self.mlp(x)
```

**Comment by rusty1s** [comment]
All you need to change is to replace the `node_emb` module with a `Linear` layer, e.g., `Linear(5, channels - pe_dim)`. Let me know :)

**Comment by kou18n** [comment]
> All you need to change is to replace the `node_emb` module with a `Linear` layer, e.g., `Linear(5, channels - pe_dim)`. Let me know :)

Thank you for your reply. 
It's works.
One more question, currently, I made a dummy edge_attr with 1D. 
Can GraphGPS not use edge features?
```python
conv = GPSConv(channels, GINEConv(nn), heads=4, attn_dropout=0.5)
````
it's possible to use GINConv instead?
Thank you!!!

**Comment by rusty1s** [comment]
`GraphGPS` can operate on both edge features and no edge features - it depends on which GNN layer you input to it. As such, swapping out `GINEConv` with `GINConv` should do the trick.

**Comment by kou18n** [comment]
> `GraphGPS` can operate on both edge features and no edge features - it depends on which GNN layer you input to it. As such, swapping out `GINEConv` with `GINConv` should do the trick.

Thank you very much. The example code is a regression, I want to do the classification and calculate the accuracy metric. 
How should I modify the code? I already tried  
` pred = model(data.x, data.pe, data.edge_index, data.edge_attr, data.batch).argmax(dim=-1)`  and 
`pred = model(data.x, data.pe, data.edge_index, data.edge_attr,data.batch).squeeze()`
It doesn't work.

```python
@torch.no_grad()
def test(loader):
    model.eval()

    total_correct = 0
    for data in loader:
        data = data.to(device)
        pred = model(data.x, data.pe, data.edge_index, data.edge_attr,
                    data.batch).squeeze() #????
        total_correct += int((pred == data.y).sum())
    return total_correct / len(loader.dataset)
```
